Xamarin.Forms.Platform.Android/CollectionView/ObservableItemsSource.cs

```diff
@@ -49,7 +49,17 @@ void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 
 		void Move(NotifyCollectionChangedEventArgs args)
 		{
-			_adapter.NotifyItemMoved(args.OldStartingIndex, args.NewStartingIndex);
+			var count = args.NewItems.Count;
+
+			if (count == 1)
+			{
+				// For a single item, we can use NotifyItemMoved and get the animation
+				_adapter.NotifyItemMoved(args.OldStartingIndex, args.NewStartingIndex);
+				return;
+			}
+
+			_adapter.NotifyItemRangeChanged(args.OldStartingIndex, count);
+			_adapter.NotifyItemRangeChanged(args.NewStartingIndex, count);
```

Comment:
Also need notify items between `OldStartingIndex + count`  and `NewStartingIndex`.
http://recordit.co/088QU2waIn
Perhaps you can fix it like this
```C#
var start = Math.Min(args.OldStartingIndex, args.NewStartingIndex);
var end = Math.Max(args.OldStartingIndex, args.NewStartingIndex) + count;
_adapter.NotifyItemRangeChanged(start, end);
```

Reply:
Excellent point.

Reply:
Okay, should be fixed now.